spec/unit/configurer/plugin_handler_spec.rb

```diff
@@ -8,33 +8,84 @@
 
-  before :each do
-    # PluginHandler#load_plugin has an extra-strong rescue clause
-    # this mock is to make sure that we don't silently ignore errors
-    Puppet.expects(:err).never
+
+  context "server agent version is 5.3.4" do
+
+    before :each do
+      # PluginHandler#load_plugin has an extra-strong rescue clause
+      # this mock is to make sure that we don't silently ignore errors
+      Puppet.expects(:err).never
+      # Server_agent version needs to be at 5.3.4 in order to mount locales
+      Puppet.push_context({:server_agent_version => "5.3.4"})
+    end
+
+    it "downloads plugins, facts, and locales" do
+      plugin_downloader = stub('plugin-downloader', :evaluate => [])
+      facts_downloader = stub('facts-downloader', :evaluate => [])
+      locales_downloader = stub('locales-downloader', :evaluate => [])
+
+      factory.expects(:create_plugin_downloader).returns(plugin_downloader)
+      factory.expects(:create_plugin_facts_downloader).returns(facts_downloader)
+      factory.expects(:create_locales_downloader).returns(locales_downloader)
+
+      pluginhandler.download_plugins(environment)
+    end
+
+    it "returns downloaded plugin, fact, and locale filenames" do
+      plugin_downloader = stub('plugin-downloader', :evaluate => %w[/a])
+      facts_downloader = stub('facts-downloader', :evaluate => %w[/b])
+      locales_downloader = stub('locales-downloader', :evaluate => %w[/c])
+
+      factory.expects(:create_plugin_downloader).returns(plugin_downloader)
+      factory.expects(:create_plugin_facts_downloader).returns(facts_downloader)
+      factory.expects(:create_locales_downloader).returns(locales_downloader)
+
+      expect(pluginhandler.download_plugins(environment)).to match_array(%w[/a /b /c])
+    end
   end
 
-  it "downloads plugins, facts, and locales" do
-    plugin_downloader = stub('plugin-downloader', :evaluate => [])
-    facts_downloader = stub('facts-downloader', :evaluate => [])
-    locales_downloader = stub('locales-downloader', :evaluate => [])
+  context "server agent version is 5.3.3" do
+
+    before :each do
+      # PluginHandler#load_plugin has an extra-strong rescue clause
+      # this mock is to make sure that we don't silently ignore errors
+      Puppet.expects(:err).never
+      # Server_agent version needs to be at 5.3.4 in order to mount locales
+      Puppet.push_context({:server_agent_version => "5.3.3"})
+    end
 
-    factory.expects(:create_plugin_downloader).returns(plugin_downloader)
-    factory.expects(:create_plugin_facts_downloader).returns(facts_downloader)
-    factory.expects(:create_locales_downloader).returns(locales_downloader)
+    it "returns downloaded plugin, fact, and but not locale filenames" do
+      plugin_downloader = stub('plugin-downloader', :evaluate => %w[/a])
+      facts_downloader = stub('facts-downloader', :evaluate => %w[/b])
+      locales_downloader = stub('locales-downloader', :evaluate => %w[/c])
```

Comment:
We shouldn't need the locales_downloader

Reply:
My thinking here was that if the code got corrupted and the test failed we would get something like "expected: [/a, /b] but got [/a, /b, /c]" in the case that the code incorrectly started trying to download locales. Let's review tomorrow if you're available.